operator/v1/types_ingress.go

```diff
@@ -1346,6 +1346,75 @@ type IngressControllerHTTPHeaders struct {
 	// +nullable
 	// +optional
 	HeaderNameCaseAdjustments []IngressControllerHTTPHeaderNameCaseAdjustment `json:"headerNameCaseAdjustments,omitempty"`
+
+	// headersManipulation specifies options for performing action on headers.
+	// Currently one either perform a Set or Delete operation on HTTP Response headers.
+	// For Setting Headers you can provide a value for the header whereas for deleting one needs to provide header name.
+	// +kubebuilder:validation:Optional
+	HeadersManipulation IngressControllerHTTPHeadersManipulation `json:"headersManipulation,omitempty"`
+}
+
+// IngressControllerHeadersManipulation defines whether we want to set/delete Response or Request Headers.
+// Currently set/delete of HTTP Response Headers is available only.
+type IngressControllerHTTPHeadersManipulation struct {
+	// Response is a list of HTTP Headers which either need to set or deleted.
+	Response []IngressControllerSetOrDeleteHTTPHeader `json:"response,omitempty"`
+}
+
+// IngressControllerSetOrDeleteHTTPHeader specifies options for set/delete a header.
+type IngressControllerSetOrDeleteHTTPHeader struct {
+	// name specifies a header name to be set/delete.  Its value must be a valid HTTP header.
+	// name as defined in RFC 2616 section 4.2.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:Pattern="^[-!#$%&'*+.0-9A-Z^_`a-z|~]+$"
+	Name string `json:"name"`
+
+	// value specifies fields to mention the header details and the action you want to perform on them.
+	Value IngressControllerSetOrDeleteHTTPHeaderUnion `json:"value"`
+}
+
+// IngressControllerSetOrDeleteHTTPHeaderUnion specifies fields to mention the header details and the action you want to perform on them.
+// +union
+type IngressControllerSetOrDeleteHTTPHeaderUnion struct {
+	// action defines the type of the action to be applied on the header.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:=Set;Delete
+	// +kubebuilder:validation:Required
+	Action IngressControllerHTTPHeaderAction `json:"action,omitempty"`
+
+	// set defines HTTP headers that should be set.
+	// If this field is empty, no headers are added.
+	//
+	// Note that this option only applies to cleartext HTTP connections
+	// and to secure HTTP connections for which the ingress controller
+	// terminates encryption (that is, edge-terminated or reencrypt
+	// connections).  Headers cannot be set for TLS passthrough
+	// connections.
+	//
+	// +kubebuilder:validation:Optional
+	// +unionMember
+	Set *IngressControllerSetHTTPHeader `json:"set,omitempty"`
```

Comment:
Does this need to be a struct? Do you ever see this having anything within it over than the value string?

Reply:
It might be better to have a struct. I am not sure if in the future we might need more values in it. cc @Miciah

Reply:
The only thing I can think of offhand is that we could add a field for specifying conditions, but conditionals are supported for `del-header` as well as `set-header`, so it might not make sense as an action-specific option anyway.

Reply:
Having it as a struct complicates things a little, but does make it more expandable in the future, I'm ok with keeping it if we are happy we've got the naming of the various levels of the unions correct now